### ISSUE TITLE
MainForm.cs not opening

### ISSUE THREAD

**Issue by Suraj2093** [microsoft/MCW-Internet-of-Things#26]
In Hands-on-lab, In Exercise 1 Task 2 the MainForm.cs in SmartMeterSimulator.sln is not opening up. Seems like there is some error in the MainForm.Designer.cs. Please look into this

Also the Virtual Machine Image **Visual Studio Community 2019 (latest) on Windows Server 2016 (x64)** mentioned in Before Hands-on-Lab is currently not available in the Azure Marketplace

@DawnmarieDesJardins Please check these issues

**Comment by timahenning** [comment]
We are looking into this issue now.

Tim

**Comment by TheTrainingBoss** [comment]
I can't reproduce the MainForm.cs not opening in VS 2019
![MainForm](https://user-images.githubusercontent.com/52613206/69484011-1ac10b00-0dfc-11ea-906a-0f046c408dda.jpg)

**Comment by TheTrainingBoss** [comment]
The option on Visual Studio 2019 Community just changed from Windows 2016 to 2019 and will be adjusted during the next revision of this MCW.  for now please use the one that uses Windows 2019.

![VSCommunity](https://user-images.githubusercontent.com/52613206/69484064-ca967880-0dfc-11ea-826c-d0dcb4c7ab46.jpg)

**Comment by timahenning** [comment]
We are closing this issue.  Please let us know if you have additional questions and we can re-open it.

**Comment by Suraj2093** [comment]
I am still facing the same issue, seems like it is because DeviceManager.cs is incomplete before opening MainForm.cs

![image](https://user-images.githubusercontent.com/30050847/70047483-28da0d00-15ee-11ea-9e1b-e70e664abdf1.png)

![image](https://user-images.githubusercontent.com/30050847/70047648-70609900-15ee-11ea-9dc9-31621b5aa96c.png)

**Comment by TheTrainingBoss** [comment]
Yes, please follow the instructions in the MCW to uncomment  the TODO lines in the code and the form and project will work with no issues.  I just tried it again from a clean repo and everything worked if you follow the steps in the MCW in correct order without skipping any steps.  Thanks.

**Comment by timahenning** [comment]
@Suraj2093 Are saying you cannot open the Form in the VS Designer because there is a dependency on the DeviceManager.cs being complete?  Please try completing the DeviceManager.cs (Exercise 2, Task 1) first and then go back and try to open the form.  Let me know if that works.

**Comment by timahenning** [comment]
@Suraj2093 I am closing this issue for now.  If you have additional debugging information or questions, we can reopen the issue.

**Comment by Suraj2093** [comment]
@timahenning I was able to go through this step after making changes in DeviceManager.cs.

Thanks

**Comment by timahenning** [comment]
@Suraj2093 I am going to make a suggestion to the author about switching the steps around.  Thank you for bringing this to our attention and responding back the solution fixed the problem.

**Comment by SeeratArora** [comment]
Hi @timahenning ,
In [Exercise 1: Task 2:](https://github.com/microsoft/MCW-Internet-of-Things/blob/master/Hands-on%20lab/HOL%20step-by-step%20-%20Internet%20of%20Things.md#task-2-configure-the-smart-meter-simulator) MainForm.cs file does not open and displays error message shown below:

![image](https://user-images.githubusercontent.com/50323628/73162529-fba06b80-4113-11ea-9906-c13cc614fd34.png)

We tried performing [Exercise 2: Task 3](https://github.com/microsoft/MCW-Internet-of-Things/blob/master/Hands-on%20lab/HOL%20step-by-step%20-%20Internet%20of%20Things.md#task-3-verify-device-registration-and-telemetry) till only Step 1 then performed [Exercise 1](https://github.com/microsoft/MCW-Internet-of-Things/blob/master/Hands-on%20lab/HOL%20step-by-step%20-%20Internet%20of%20Things.md#exercise-1-iot-hub-provisioning) and continued with Exercise 2: Task 3: Step 2.

We got proper results.

Can you please check whether we steps need to be shuffled or there's some another issue. Please let me know.

**Comment by SeeratArora** [comment]
Hi @timahenning 
Can you please let me know updates on this?

**Comment by DawnmarieDesJardins** [comment]
@SeeratArora 
Hello, please see the above conversation, making changes in DeviceManager.cs should allow you to complete the lab.

**Comment by mtochs** [comment]
I'm having the same issues with this lab and do not see a solution listed above other than starting over.  Is there an alternative, smoother way to create a simulator for this lab?

**Comment by timahenning** [comment]
Reviewing the issue.

**Comment by timahenning** [comment]
The lab author says he is not seeing the issue.  Can double check your steps?

**Comment by mtochs** [comment]
Yes I tried three times.  Are you able try it?  Also is there an alternative available?  Editing the lines of C# code opens the lab up to error and doesn't appear to add to the learning experience.

**Comment by TheTrainingBoss** [comment]
Thanks mtochs, I tried twice on 2 different machines using Visual Studio 2019 and when I follow the steps in the HOL step by step, everything works fine.  Also this lab has been verified by others and I teach classes online and onsite based on these labs in this MCW.  The error occurs if you try to open the form in the IDE before completing the TODO changes which means you did not follow the steps in the lab.  In any case, my suggestion if you feel the editing of C# code does not add learning experience is to copy and past the code for the DeviceManager.cs from the lab all at once and replace the current file instead of going through the Todo List one by one and the file will open right away with no errors as well.  Hope that helps.
If Microsoft would like me to remove the ToDo list in future revisions and just place the completed code by default, I will sure do that in the next revision.  When I inherited this lab, this is the way Microsoft wanted the lab presented to make sure that people understand the important parts of coding this lab and completing them on their own if they can or go through the step by step to get a clearer picture of what needs to be done.
Thanks
